core/src/main/java/com/aem/demo/core/models/MainPageModel.java

```diff
@@ -100,4 +106,27 @@ private List<NewsCardModel> separateModels(List<NewsCardModel> list, int paginat
         return separatedPart;
     }
 
+    private void paginateModels(List<NewsCardModel> models) {
+        double listSize = models.size();
+        double listPart = listSize / 5;
+        double roundPart = Math.ceil(listPart);
+
+        pageNumbers = IntStream.range(1, (int) (roundPart + 1)).boxed().collect(Collectors.toList());
```

Comment:
As far as can see, if `models.size() == 4`, there will be two pages,
`listPart 4.0 / 5.0 = 0,8`
`roundPart = Math.ceil(0.8) = 1.0`
`(int) roundPart +1  = 2`

Therefore, when we use `Math.ceil` we do not increment the result any more.
Please @Zubayr99 always try to think one step ahead, like in a chess game. Try to see the situation in general. Try to imagine, what are the consequences of any change you have introduced